conan/tools/microsoft/toolchain.py

```diff
@@ -49,20 +51,59 @@ def generate(self):
         self._write_config_toolchain(config_filename)
         self._write_main_toolchain(config_filename, condition)
 
+    @staticmethod
+    def _msvs_toolset(settings):
+        compiler = settings.get_safe("compiler")
+        compiler_version = settings.get_safe("compiler.version")
+        if compiler == "msvc":
+            version = compiler_version[:4]  # Remove the latest version number 19.1X if existing
+            toolsets = {'19.0': 'v140',  # TODO: This is common to CMake, refactor
+                        '19.1': 'v141',
+                        '19.2': 'v142'}
+            return toolsets[version]
+        if compiler == "intel":
+            compiler_version = compiler_version if "." in compiler_version else \
+                "%s.0" % compiler_version
+            return "Intel C++ Compiler " + compiler_version
+        if compiler == "Visual Studio":
+            toolset = settings.get_safe("compiler.toolset")
+            if not toolset:
+                toolsets = {"16": "v142",
+                            "15": "v141",
+                            "14": "v140",
+                            "12": "v120",
+                            "11": "v110",
+                            "10": "v100",
+                            "9": "v90",
+                            "8": "v80"}
+                toolset = toolsets.get(compiler_version)
+            return toolset or ""
+
+    @staticmethod
+    def _runtime_library(settings):
+        compiler = settings.compiler
+        runtime = settings.get_safe("compiler.runtime")
+        if compiler == "msvc":
+            build_type = settings.get_safe("build_type")
+            if build_type != "Debug":
+                runtime_library = {"static": "MultiThreaded",
+                                   "dyanmic": "MultiThreadedDLL"}.get(runtime, "")
+            else:
+                runtime_library = {"static": "MultiThreadedDebug",
+                                   "dyanmic": "MultiThreadedDebugDLL"}.get(runtime, "")
```

Comment:
```suggestion
                                   "dynamic": "MultiThreadedDebugDLL"}.get(runtime, "")
```

Reply:
Good catch! 

I have added an integration test, much faster than functional ones, that is capturing this error. So far the effort was using only functional tests for the big picture working, but it is needing also some integration ones.